scripts/monitoring/auto_scale_workers.py

```diff
@@ -0,0 +1,256 @@
+    return response.json()
+
+
+def get_challenges():
+    all_challenge_endpoint = (
+        "{}/api/challenges/challenge/all/unapproved/all".format(
+            evalai_endpoint  # Gets all unapproved challenges
+        )
+    )
+    response = execute_get_request(all_challenge_endpoint)
+
+    return response.json()
+
+
+# TODO: Check if we can use challenge["queue"] directly to get queue name.
+# TODO: Understand if all the attributes are exposed in the API calls, or are they done selectively.
+def get_sqs_queue_by_name(queue_name):
+    """
+    Returns:
+        Returns the SQS Queue object
+    """
+    if settings.DEBUG or settings.TEST:
+        sqs = boto3.resource(
+            "sqs",
+            endpoint_url=os.environ.get("AWS_SQS_ENDPOINT", "http://sqs:9324"),
+            region_name=os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+            aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
+            aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
+        )
+    else:
+        sqs = boto3.resource(
+            "sqs",
+            region_name=os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+            aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
+            aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
+        )
+    if queue_name == "":
+        queue_name = "evalai_submission_queue"
+    # Check if the queue exists. If no, then create one
+    try:
+        queue = sqs.get_queue_by_name(QueueName=queue_name)
+    except botocore.exceptions.ClientError as ex:
+        if (
+            ex.response["Error"]["Code"]
+            != "AWS.SimpleQueueService.NonExistentQueue"
+        ):
+            raise Exception("Cannot get queue: {}".format(queue_name))
+    return queue
+
+
+def get_queue_name(param, challenge_pk):
+    """
+    Generate unique SQS queue name of max length 80 for a challenge
+
+    Arguments:
+        param {string} -- challenge title
+        challenge_pk {int} -- challenge primary key
+
+    Returns:
+        {string} -- unique queue name
+    """
+    # The max-length for queue-name is 80 in SQS
+    max_len = 80
+    max_challenge_title_len = 50
+
+    env = settings.ENVIRONMENT
+    queue_name = param.replace(" ", "-").lower()[:max_challenge_title_len]
+    queue_name = re.sub(r"\W+", "-", queue_name)
+
+    queue_name = "{}-{}-{}-{}".format(
+        queue_name, challenge_pk, env, uuid.uuid4()
+    )[:max_len]
+    return queue_name
+
+
+def get_queue_name_by_challenge(challenge):
+    challenge_id = challenge["id"]
+    challenge_title = challenge["title"]
+    return get_queue_name(challenge_id, challenge_title)
+
+
+def get_queue_length(queue):
+    # TODO: Check if we should use approximate number of messages
+    # https://github.com/boto/boto3/issues/599
+    # Ref: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/sqs.html#using-an-existing-queue
+    count = 0
+    for message in queue.receive_messages():
```

Comment:
isn't there a `count()` API in boo3?

Reply:
The methods listed here do not contain a `count()` method: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sqs.html
I also checked some stackoverflow answers regarding this. Will look up once again.